File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) {
    let answer =0;
    answer =regration(num,answer);
    return answer;
}

function regration(num,count){
    if(num===1){
        return count;
    }
    if(count===500){
        return -1;
    }
    if(num%2===0){
        count+=1;
        num/=2;
    }else{
        count+=1;
        num= (num*3)+1;
    }
    return regration(num,count);
}

File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public long solution(int num) {
        long answer = recursion(num,0);
        return answer;
    }
    
    public long recursion(long n, int count){
        if(count>=500){
            return -1;
        }
        if(n==1){
            return count;
        }
        if(n%2==0){
            return recursion(n/2,count+1);
        }else{
            return recursion(n*3+1,count+1);
        }
    }
}